### PR TITLE
Set boost for a job

### PR DESCRIPTION
**Pull request by g4s8**
#184 - set boost for a job
(from #395)

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#459](https://github.com/zerocracy/farm/issues/459) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#459](https://github.com/zerocracy/farm/issues/459) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by g4s8** [comment]
@fabriciofx you have already reviewed this PR in #395, but my local fork was missed by Github when "farm" project becomes public, could you please take a look?

**Comment by 0crat** [comment]
The user @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#459](https://github.com/zerocracy/farm/issues/459), please stop working

**Comment by 0crat** [comment]
This pull request [#459](https://github.com/zerocracy/farm/issues/459) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@yegor256 should be assigned to @fabriciofx I think

**Comment by 0crat** [comment]
The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#459](https://github.com/zerocracy/farm/issues/459), please stop working

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/459#issuecomment-364914314))

@g4s8 This is what I know about this job, as in [§32](http://datum.zerocracy.com/pages/policy.html#32):

  * The job [#459](https://github.com/zerocracy/farm/issues/459) is [in scope](http://datum.zerocracy.com/pages/policy.html#14) for 18days
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://datum.zerocracy.com/pages/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
    * @fabriciofx[/z](https://www.0crat.com/u/fabriciofx): User was resigned from the ticket
    * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda): User was resigned from the ticket

File: src/main/resources/com/zerocracy/stk/pm/staff/boost/set_boost.groovy [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.zerocracy.farm.Assume
+import com.zerocracy.pm.ClaimIn
+import com.zerocracy.pm.cost.Boosts
+
+def exec(Project project, XML xml) {
+  new Assume(project, xml).notPmo()
+  new Assume(project, xml).type('Set boost')
+  def claim = new ClaimIn(xml)
+  def factor = Integer.valueOf(claim.param('factor')
+    .replaceAll('x$', ''))
```
Comment (yegor256) [inline src/main/resources/com/zerocracy/stk/pm/staff/boost/set_boost.groovy]: @g4s8 what is the point of using two lines here? Keep in mind that the line length limit in Groovy is 120

File: src/main/resources/com/zerocracy/stk/pm/staff/boost/set_boost.groovy [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  new Boosts(project).bootstrap()
+    .boost(claim.param('job'), factor)
```
Comment (yegor256) [inline src/main/resources/com/zerocracy/stk/pm/staff/boost/set_boost.groovy]: @g4s8 same here, put it all in one line

**Comment by yegor256** [comment]
@g4s8 just two small comments above

**Comment by g4s8** [comment]
@yegor256 got it, please look again.

**Comment by yegor256** [comment]
@rultor merge

**Comment by g4s8** [comment]
@yegor256 please restart merge, should be fine after #549 fix

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:zerocracy/farm#459` is not assigned, can't get start time

**Comment by 0crat** [comment]
The job [#459](https://github.com/zerocracy/farm/issues/459) is now out of scope

**Pull request merged**